experiments/modern/src/maki-interpreter/virtualMachine.js

```diff
@@ -134,16 +144,34 @@ async function interpret(start, program, { logger = null }) {
         stack.push(obj[methodName](...methodArgs));
         break;
       }
+      // callGlobal
+      case 25: {
+        // This is where the version checked wa5 scripts start with this offset
+        if (command.arguments[0].offset === 4294967296) {
+          // skip ahead to where the real program begins
+          i = i + 3;
+          break;
+        }
+        const offset = command.arguments[0].offset;
+        const nextCommandIndex = offsetToCommand[offset];
+        const value = await interpret(nextCommandIndex, program, stack, { logger });
```

Comment:
Ugh, this `await` in here (and all the other places) is killing me. I really want to find a better way to handle debug mode, because I suspect this might slow things down in non-debug mode.

Okay to leave it for now. I'll start a list of things I want to come back and clean up. I like the idea of just getting it working (with tests!) and then being able to come back and clean things up after.

Reply:
Yea, its really only here and then at the top level of a few other things, will be easy enough to rip out if we can get rid of the async.

Yea, definitely gonna a need some cleanup, getting a bit hacky in some places hah